login/src/main/java/com/fightpandemics/login/ui/profile/CompleteProfileFragment.kt

```diff
@@ -6,102 +6,165 @@ import android.os.Bundle
 import com.fightpandemics.core.data.model.login.CompleteProfileRequest
 import com.fightpandemics.core.data.model.login.Hide
-import com.fightpandemics.core.data.model.login.Location
 import com.fightpandemics.core.data.model.login.Needs
 import com.fightpandemics.core.data.model.login.Objectives
+import com.fightpandemics.core.data.model.userlocationpredictions.Prediction
 import com.fightpandemics.core.utils.ViewModelFactory
-import com.fightpandemics.login.R
+import com.fightpandemics.core.widgets.BaseLocationFragment
 import com.fightpandemics.login.dagger.inject
+import com.fightpandemics.login.databinding.FragmentCompleteProfileBinding
 import com.fightpandemics.login.ui.LoginViewModel
+import com.fightpandemics.login.util.dismissKeyboard
 import com.fightpandemics.login.util.snack
 import com.google.android.material.appbar.MaterialToolbar
-import com.google.android.material.checkbox.MaterialCheckBox
 import com.google.android.material.snackbar.Snackbar
-import com.google.android.material.textfield.TextInputEditText
+import kotlinx.android.synthetic.main.complete_profile_location.etAddress
+import kotlinx.android.synthetic.main.complete_profile_location.view.auto_complete_locations_recycler_view
+import kotlinx.android.synthetic.main.complete_profile_location.view.etAddress
+import kotlinx.android.synthetic.main.complete_profile_location.view.item_line_divider1
+import kotlinx.android.synthetic.main.complete_profile_location.view.progressBar
+import kotlinx.android.synthetic.main.complete_profile_location.view.share_my_location
+import kotlinx.android.synthetic.main.complete_profile_location.view.tilLocation
+import kotlinx.android.synthetic.main.donation_checkbox.view.donation_checkbox
+import kotlinx.android.synthetic.main.information_checkbox.view.information_checkbox
+import kotlinx.android.synthetic.main.medical_help_checkbox.view.hours_request_checkbox
+import kotlinx.android.synthetic.main.other_help_checkbox.view.other_help_checkbox
+import kotlinx.android.synthetic.main.volunteer_hrs_checkbox.view.hours_offer_checkbox
 import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.FlowPreview
+import kotlinx.coroutines.flow.collect
 import timber.log.Timber
 import javax.inject.Inject
 
-class CompleteProfileFragment : Fragment() {
+@ExperimentalCoroutinesApi
+@FlowPreview
+class CompleteProfileFragment : BaseLocationFragment(), LocationAdapter.OnItemClickListener {
+    private val adapter = LocationAdapter(this)
 
     @Inject
     lateinit var loginViewModelFactory: ViewModelFactory
 
-    private lateinit var completeProfileButton: ConstraintLayout
-    @ExperimentalCoroutinesApi
     private val loginViewModel: LoginViewModel by viewModels { loginViewModelFactory }
     private lateinit var completeProfileToolbar: MaterialToolbar
 
+    private var _fragmentCompleteProfileBinding: FragmentCompleteProfileBinding? = null
+    private val fragmentCompleteProfileBinding get() = _fragmentCompleteProfileBinding!!
```

Comment:
Doesn't this crash if someone tries to access `fragmentCompleteProfileBinding`?

Reply:
It would only crash if our binding is null. This is the recommended way to initialize view bindings inside fragments. We need the type to be nullable because we need to free it on the destroy method. 

From: https://developer.android.com/topic/libraries/view-binding#fragments
Note: Fragments outlive their views. Make sure you clean up any references to the binding class instance in the fragment's onDestroyView() method.